src/wallet/wallet.ts

```diff
@@ -143,16 +186,45 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
    * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async start() {
+  async start({ pinCode }) {
     if (!this.seed) {
       throw new Error('Seed should be in memory when starting the wallet.');
     }
 
     this.setState(walletState.LOADING);
+
+    const xpriv = walletUtils.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
     const xpub = walletUtils.getXPubKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
+    const authXpub = HathorWalletServiceWallet.getAuthXPubKeyFromSeed(this.seed, {
+      passphrase: this.passphrase,
+      networkName: this.network.name,
+    });
+
+    const walletId = walletUtils.sha256d(xpub, 'hex');
+    const now = new Date();
+    const timestampNow = Math.floor(now.getTime() / 1000);
+
+    // prove we own the xpubkey
+    const xpubAccountPath = walletUtils.deriveXpriv(xpriv, '0\'')
+    const address = xpubAccountPath.publicKey.toAddress(this.network.getNetwork()).toString();
+    const message = new bitcore.Message(String(timestampNow).concat(walletId as string).concat(address));
+    const xpubkeySignature = message.sign(xpubAccountPath.privateKey);
+
+    // prove we own the auth_xpubkey
+    const authDerivedPrivKey = HathorWalletServiceWallet.deriveAuthXpriv(xpriv);
+    const authAddress = authDerivedPrivKey.publicKey.toAddress(this.network.getNetwork());
+
+    const authMessage = new bitcore.Message(String(timestampNow).concat(walletId as string).concat(authAddress));
+    const authXpubkeySignature = authMessage.sign(authDerivedPrivKey.privateKey);
```

Comment:
Maybe we can create a method to get the signature (I think we already has this method but we should only refactor it). The method would receive the `xpriv`, `timestamp` and `walletId`, then it can create the address, create the message and return the signature.

This would prevent differences in the code as the onde mentioned by @luislhl in the comment above.

Reply:
I just checked and we can use the method `signMessage` that already exists. You don't even need to send `walletId`, given that it uses from `this`.

Reply:
You are absolutely right, sorry for that

Refactored the `signMessage` method to accept `walletId` and refactored the `start` method to use it instead of manually signing messages